src/adapters/terms_contract_parameters.ts

```diff
@@ -0,0 +1,18 @@
+import { BigNumber } from "utils/bignumber";
+
+export namespace TermsContractParameters {
+    export function generateHexValueOfLength(length: number): string {
+        return "0x" + "f".repeat(length);
+    }
+
+    export function bitShiftLeft(target: BigNumber, numPlaces: number): BigNumber {
+        const binaryTargetString = target.toString(2);
+        const binaryTargetStringShifted = binaryTargetString + "0".repeat(numPlaces);
+        return new BigNumber(binaryTargetStringShifted, 2);
+    }
+
+    export function isDecimalValue(value: BigNumber): boolean {
+        const [, rightOfDecimal] = value.toString().split(".");
+        return rightOfDecimal.length > 0;
```

Comment:
Careful, as I'm pretty sure this will fail if `rightOfDecimal` ends up being undefined (i.e. if there is no decimal place)

Reply:
Great catch! Fixing.